### PR TITLE
Fixed `String>>#keywords`

### PR DESCRIPTION
**Pull request by Mathilde411**
Fixed problem from pharo-project/pharo#14140

Added comment and tests for method `String>>#keywords`.
Created `Symbol>>#keywordsStrict` and tests for new usages.

The final goal is to deprecate `String>>#keywords`.

### CODE REVIEW

File: src/Collections-Strings-Tests/StringTest.class.st [hunk 1]
```diff
@@ -1461,6 +1461,14 @@ StringTest >> testIsWideString [
 	self deny: 'éàôüÖ' isWideString
 ]
 
+{ #category : #tests }
+StringTest >> testKeywords [
+
+	self assert: (#foo: keywords) equals: #('foo:').
+	self assert: (#foo:bar: keywords) equals: #('foo:' 'bar:').
```
Comment (Ducasse) [inline src/Collections-Strings-Tests/StringTest.class.st:1468]: The test should also show the case for unary and binary messages

File: src/Collections-Strings/Symbol.class.st [hunk 2]
```diff
@@ -495,6 +495,15 @@ Symbol >> isUnary [
 	^ self precedence = 1
 ]
 
+{ #category : #sorting }
+Symbol >> keywordsStrict [
+	"Returns the keywords of the provided selector. Does not assume that the reciever is keyword-based."
+
```
Comment (Ducasse) [inline src/Collections-Strings/Symbol.class.st:509]: reciever -> receiver

Please add also some executable comments.
I could not understand what is does not assume. So the examples will be more self explanatory

File: src/Collections-Strings-Tests/StringTest.class.st (hunk 1 above)
Comment (Mathilde411) [reply src/Collections-Strings-Tests/StringTest.class.st:1468]: This is the test for the method that should only accept keyword-based selectors, so there shouldn't be any need to test

**Comment by guillep** [comment]
I think the original issue from @Ducasse was to have a method that answers you every selector part that accepts an argument, that for unary/binary returns an empty collection. In that case, it would be expected to have a test asserting that it returns an empty collection (or not in the non strict case), right?

**Comment by Ducasse** [comment]
agreed!

**Pull request merged**